table/tables/tables_test.go

```diff
@@ -943,3 +943,93 @@ func TestTxnAssertion(t *testing.T) {
 	testUntouchedIndexImpl("OFF", false)
 	testUntouchedIndexImpl("OFF", true)
 }
+
+func TestInsertNotLock(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+	tk2 := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk2.MustExec("use test")
+	tk.MustExec("create table t(id int primary key, v int)")
+
+	tk.MustExec("set @@tidb_constraint_check_in_place_pessimistic = 0")
+	tk.MustExec("begin pessimistic")
+	tk.MustExec("insert into t values(1, 0)")
+	tk2.MustExec("begin pessimistic")
+	tk2.MustExec("insert into t values (1, 1)")
+	tk2.MustExec("commit")
+	_, err := tk.Exec("commit")
+	require.NotNil(t, err)
+
+	tk.MustQuery("select * from t").Check(testkit.Rows("1 1"))
+
+	tk.MustExec("admin check table t")
+}
+
+func TestDeferConstraintCheck(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+	tk2 := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk2.MustExec("use test")
+	tk.MustExec("create table t(id int primary key, v int)")
+	tk.MustExec("set @@tidb_constraint_check_in_place_pessimistic=0")
+
+	// TiKV will perform a constraint check before reporting assertion failure.
+	// And constraint violation precedes assertion failure.
+	if *testkit.WithTiKV == "" {
+		tk.MustExec("set @@tidb_txn_assertion_level=off")
```

Comment:
Since constraint violation precedes assertion failure on TiKV, why do we close `tidb_txn_assertion_level`?  It always reports an error because of `constraint violation` whether  `tidb_txn_assertion_level` is on or off

Reply:
This is for unistore test. It has a different assertion implementation